### PR TITLE
Adding cmdlets for managing MachineLearning CommitmentPlans ARM resources

### PR DESCRIPTION
**Pull request by ericwill-msft**
## Comments

---

This checklist is used to make sure that common issues in a pull request are covered by the creator. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).

Below in **Overall Changes**, check off the boxes that apply to your PR. For the categories that you did not check off, you can remove them from this body. Within each of the categories that you did select, make sure that you can check off **all** of the boxes. 

For information on cleaning up the commits in your pull request, click [here](../documentation/cleaning-up-commits.md).
## Overall Changes
- [x] [**MANDATORY** - General changes](#general)
- [x] [**MANDATORY** - Add/remove/edit test(s)](#tests)
- [x] [Add/remove/edit cmdlet(s)](#cmdlet-signature)
### General
- [x] Title of the PR is clear and informative
- [x] There are a small number of commits that each have an informative message
- [N/A] If it applies, references the bug/issue that the PR fixes
- [X] All files have the Microsoft copyright header
- [x] Cmdlets refer to management libraries through nuget references - no dlls are checked in
- [x] The PR does not introduce breaking changes (unless a major version change occurs in the assembly and module)
### Tests
- [x] PR includes test coverage for the included changes
- [x] Tests must use xunit, and should either use Moq to mock management client calls, or use the scenario test framework
- [x] PowerShell scripts used in tests must not use hard-coded values for location
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for existing resources
- [x] Tests should not use App.config files for settings
- [x] Tests should use the built-in PowerShell functions for generating random names when unique names are necessary - this will store names in the test recording
- [N/A] Tests should use Start-Sleep to pause rather than Thread.Sleep
### Cmdlet Signature
- [x] Cmdlet name uses an approved PowerShell verb - use the enums for `VerbsCommon`, `VerbsCommunication`, `VerbsLifecycle`, `VerbsOther` whenever possible
- [x] Cmdlet noun name uses the AzureRm prefix for management cmdlets, and the Azure prefix for data plane cmdlets
- [x] Cmdlet specifies the `OutputType` attribute if any output is produced; if the cmdlet produces no output, it should implement a `PassThrough` parameter
- [x] If the cmdlet makes changes or has side effects, it should implement `ShouldProcess` and have `SupportShouldProcess = true` specified in the cmdlet attribute. See a discussion about correct `ShouldProcess` implementation [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change)
- [x] Cmdlets should derive from AzureRmCmdlet for management cmdlets, and AzureDataCmdlet for data cmdlets
- [N/A] If multiple parameter sets are implemented, the cmdlet should specify a `DefaultParameterSetName` in its cmdlet attribute

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ericwill-msft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ericwill). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/CommitmentPlanTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+function Test-CreateGetRemoveMLCommitmentPlan
+{  
+    $planDeleted = $false
+
+    $actualTest = {
+        param([string] $location)
+
+		try
+		{
+			$resourceGroupName = Get-ResourceGroupName
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/CommitmentPlanTests.ps1]: @ericwill-msft it looks like there are some indentation issues with this file

File: src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/MoveAzureMLCommitmentAssociation.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Management.MachineLearning.CommitmentPlans.Models;
+
+namespace Microsoft.Azure.Commands.MachineLearning
+{
+    [Cmdlet(VerbsCommon.Move, CommitmentPlansCmdletBase.CommitmentAssociationCommandletSuffix)]
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/MoveAzureMLCommitmentAssociation.cs]: @ericwill-msft does this cmdlet make changes or have side effects? Consider implementing `ShouldProcess` like you do in the cmdlets below

File: src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/NewAzureMLCommitmentPlan.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        /// <summary>
+        /// Gets or sets a value that indicates if the user should be prompted for confirmation.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Do not ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        protected override void RunCmdlet()
+        {
+            this.ConfirmAction(
+                this.Force.IsPresent,
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/NewAzureMLCommitmentPlan.cs]: @ericwill-msft can this cmdlet override an existing `CommitmentPlan`? If not, I do not believe that `Force` is necessary

**Comment by cormacpayne** [comment]
@ericwill-msft add to whitelist

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
@ericwill-msft This PR is too late for the Ignite release. We will consider it for the next release in November

File: src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/NewAzureMLCommitmentPlan.cs (hunk 3 above)
Comment (ericwill-msft) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/NewAzureMLCommitmentPlan.cs]: This cmdlet does indeed have the ability to overwrite, if used in that way. Will leave `Force` in for now.

File: src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/MoveAzureMLCommitmentAssociation.cs (hunk 2 above)
Comment (ericwill-msft) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/MoveAzureMLCommitmentAssociation.cs]: It does have side effects. Adding `ShouldProcess`.

File: src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/CommitmentPlanTests.ps1 (hunk 1 above)
Comment (ericwill-msft) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/CommitmentPlanTests.ps1]: Looks like we're mixing tabs and spaces. Normalizing to all spaces.

**Comment by shahabhijeet** [comment]
@ericwill-msft can you please address the issues raised as well as CI build failures. No activity in last 6 days. We will close this PR if not address early next week.

**Comment by cormacpayne** [comment]
@ericwill-msft Hey Eric, the build is failing because you are missing help for the cmdlets that you added

**Comment by markcowl** [comment]
@ericwill-msft PR is failing due to missing help.  We will have to close thsi week if you cannot correct.

**Comment by shahabhijeet** [comment]
@ericwill-msft please open your PR after you fix the issue or submit a new PR.

**Comment by azurecla** [comment]
Hi __@ericwill-msft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ericwill). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/NewAzureMLCommitmentPlan.cs (hunk 3 above)
Comment (ericwill-msft) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/CommitmentPlans/NewAzureMLCommitmentPlan.cs]: OK, assuming no change required here.

**Comment by cormacpayne** [comment]
@ericwill-msft Hey Eric, only one comment: the `app.config` file, and the reference to it in the `Commands.MachineLearning.Test.csproj` file should be removed.

LGTM once that is fixed

**Pull request merged**